assets/dev/js/editor/components/template-library/manager.js

```diff
@@ -10,14 +10,52 @@ TemplateLibraryManager = function() {
 		errorDialog,
 		layout,
 		templatesCollection,
-		defaultScreen,
+		defaultRoute = 'library/templates/pages',
 		config = {},
+		modalConfig = {},
 		screens = {},
-		startIntent = {},
 		filterTerms = {};
 
 	const initLayout = function() {
 		layout = new TemplateLibraryLayoutView( { pages: screens } );
+
+		layout.getModal().on( 'hide', function() {
+			self.isOpen = false;
+			modalConfig = {};
+
+			elementorCommon.route
```

Comment:
On hide "save-template" do not restore the state.
steps: 
- open "save as template"
- do not save.
- close.
- new section
- add from library
excepted: "templates library".
got: "save as template"